src/integration_tests/scenarios/user_discovery.rs

```diff
@@ -0,0 +1,46 @@
+use crate::integration_tests::{core::*, test_cases::user_discovery::*};
+use crate::{Whitenoise, WhitenoiseError};
+use async_trait::async_trait;
+
+pub struct UserDiscoveryScenario {
+    context: ScenarioContext,
+}
+
+impl UserDiscoveryScenario {
+    pub fn new(whitenoise: &'static Whitenoise) -> Self {
+        Self {
+            context: ScenarioContext::new(whitenoise),
+        }
+    }
+}
+
+#[async_trait]
+impl Scenario for UserDiscoveryScenario {
+    fn context(&self) -> &ScenarioContext {
+        &self.context
+    }
+
+    async fn run_scenario(&mut self) -> Result<(), WhitenoiseError> {
+        tracing::info!("=== Starting Find Or Create User Scenario ===");
```

Comment:
This is not really needed, it's handled automatically for scenarios (check out the execute method on the Scenario trait (`src/integration_tests/core/traits.rs`)

Reply:
I'm not sure if I understand correctly, cause other scenarios also have the run_scenario method... 🤔